pennylane/workflow/_capture_qnode.py

```diff
@@ -585,16 +578,22 @@ def f(x):
 
         qnode.capture_cache[cache_key] = (qfunc_jaxpr, config, out_tree)
 
+    flat_shots = tuple(qnode._shots) if qnode._shots else ()  # pylint: disable=protected-access
```

Comment:
I can see this blowing up if the shot vector is something like `(10,) * 100000`. In the `Shots` class we have dense storage of shot vectors. I guess there isn't really a way around it other than maybe using a tuple of tuples, flattening its pytree, and giving the pytree def as a kwarg to the qnode primitive. Thoughts?